### PR TITLE
[GBonWPCOM e2e tests i0] Resize the Dynamic HR in the editor

### PR DESCRIPTION
**Pull request by fullofcaffeine**
#### Changes proposed in this Pull Request

Resize the Dynamic HR block in the editor to test that resizing works well in the editor and that it renders correctly in the frontend (by manually checking the screenshots).

⚠️ (Preferably) review and test this PR only after https://github.com/Automattic/wp-calypso/pull/45794 has been merged ⚠️ 

#### Testing instructions

1. Setup your env to run Calypso E2E tests (search for "Automated end-to-end Testing") in the FG;
1. Run with `./node_modules/.bin/mocha specs/wp-calypso-gutenberg-upgrade-spec.js`*[0]
1. Wait for the tests to finish. It might take a good while since it tests everything for each test site;
1. Make sure it didn't err, if it did, it might not be an actual failure, try running that specific step again. If it continues failing, let us know.
1. Add the `Needs e2e Testing Gutenberg Edge` to this PR (already added, if necessary, remove and add again);
1. Check this branch's tests in CI: [Desktop](https://circleci.com/workflow-run/376ad2ba-2c34-4883-a731-5619c8fa5d0f), [Mobile](https://circleci.com/workflow-run/1fdee30c-7da0-4398-be7d-f17b6be5ff63).

*[0] _Make sure to cd into `test/e2e` first._

### CODE REVIEW

**Comment by fullofcaffeine** [comment]
@bsessions85 Can you have a look? All tests are passing for this one.

File: test/e2e/lib/gutenberg/blocks/dynamic-separator-block-component.js [hunk 1]
```diff
@@ -12,6 +12,47 @@ class DynamicSeparatorBlockComponent extends GutenbergBlockComponent {
...
+	 * to test the functionality of the Dynamic HR in the editor and later in the
+	 * frontend too.
+	 *
+	 * @param {number} pixels the amount of pixerls to expand the HR ruler by, horizontally.
+	 */
+	async resizeBy( pixels ) {
+		// We need to move focus away from the layout grid, or any subsequent blocks inserted will be part of it
+		const resizeHandleSelector = await this.driver.findElement(
+			By.css(
+				`div[id='${ this.blockID.slice(
```
Comment (WunderBart) [inline test/e2e/lib/gutenberg/blocks/dynamic-separator-block-component.js:27]: `div[id='${ this.blockID.slice( 1 ) }']` can be just `this.blockID` (see [this comment](https://github.com/Automattic/wp-calypso/pull/45796#discussion_r500856393)).

**Review by WunderBart — APPROVED** [review]
Nice work! No blockers here, so let's address the comments in a follow-up PR.

**Review by bsessions85 — APPROVED** [review]
This looks good!

**Comment by WunderBart** [comment]
Follow-up PR: https://github.com/Automattic/wp-calypso/pull/46275
